src/properties.rs

```diff
@@ -1,38 +1,61 @@
 use std::{collections::HashMap, io::Read, str::FromStr};
 
-use xml::{EventReader, attribute::OwnedAttribute, reader::XmlEvent};
+use xml::{attribute::OwnedAttribute, reader::XmlEvent, EventReader};
 
 use crate::{
-    error::{ParseTileError, TiledError},
+    error::TiledError,
     util::{get_attrs, parse_tag},
 };
 
 #[derive(Debug, PartialEq, Eq, Copy, Clone)]
 pub struct Color {
+    pub alpha: u8,
     pub red: u8,
     pub green: u8,
     pub blue: u8,
 }
 
 impl FromStr for Color {
-    type Err = ParseTileError;
+    type Err = ();
 
-    fn from_str(s: &str) -> Result<Color, ParseTileError> {
-        let s = if s.starts_with("#") { &s[1..] } else { s };
-        if s.len() != 6 {
-            return Err(ParseTileError::ColorError);
-        }
-        let r = u8::from_str_radix(&s[0..2], 16);
-        let g = u8::from_str_radix(&s[2..4], 16);
-        let b = u8::from_str_radix(&s[4..6], 16);
-        if r.is_ok() && g.is_ok() && b.is_ok() {
-            return Ok(Color {
-                red: r.unwrap(),
-                green: g.unwrap(),
-                blue: b.unwrap(),
-            });
+    fn from_str(s: &str) -> Result<Color, Self::Err> {
+        let s = if s.starts_with("#") {
+            &s[1..]
+        } else {
+            return Err(());
```

Comment:
Actually, taking `s` as a fallback is what we need to do, because for backwards compatibility reasons, the `trans` attribute on the `image` element leaves out the `#` but is otherwise a valid color value.

(and this is actually documented for once ;-))